File: poo/ListaDeExercicio1/src/Livro.java
public class Livro {
    private String ISBN;
    private String nome;
    private int quantidadeDeVendas;
    private double preco;

    Livro (){

    }
    Livro (String ISBN, String nome){
        this.ISBN = ISBN;
        this.nome = nome;
    }
    Livro (String nome, int quantidadeDeVendas){
        this.nome = nome;
        this.quantidadeDeVendas = quantidadeDeVendas;
    }

    public String getISBN() {
        return ISBN;
    }

    public void setISBN(String ISBN) {
        this.ISBN = ISBN;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public int getQuantidadeDeVendas() {
        return quantidadeDeVendas;
    }

    public void setQuantidadeDeVendas(int quantidadeDeVendas) {
        this.quantidadeDeVendas = quantidadeDeVendas;
    }

    public double getPreco() {
        return preco;
    }

    public void setPreco(double preco) {
        this.preco = preco;
    }

    @Override
    public boolean equals(Object obj) {
        if (obj instanceof Livro){
            Livro l = (Livro) obj;
            return this.ISBN.equals(l.getISBN());
        }
        return false;
    }

    @Override
    public String toString() {
        return "Livro{" +
                "ISBN='" + ISBN + '\'' +
                ", nome='" + nome + '\'' +
                ", quantidadeDeVendas=" + quantidadeDeVendas +
                ", preco=" + preco +
                '}';
    }
}

File: poo/python/lista1Python/Livro.py
class Livro:
    def __init__(self, ISBN, nome, quantidade_de_vendas, preco):
        self.ISBN = ISBN
        self.nome = nome
        self.quantidade_de_vendas = quantidade_de_vendas
        self.preco = preco

    def __str__(self):
        return f"ISBN: {self.ISBN}, Nome: {self.nome}, Quantidade de vendas: {self.quantidade_de_vendas}, Preço: R$ {self.preco:.2f}"

    def __eq__(self, obj: object) -> bool:
        if not isinstance(obj, Livro):
            return NotImplemented
        return self.ISBN == obj.ISBN